src/persistent_merkle_tree/Node.zig

```diff
@@ -156,10 +165,21 @@ pub const Pool = struct {
 
         try pool.preheat(pool_size);
 
+        if (comptime leak_enabled) {
+            const detector = try allocator.create(LeakDetector);
+            detector.* = LeakDetector.init(allocator, .{ .level = leak_level });
+            pool.leak_detector = detector;
+        }
+
         return pool;
     }
 
     pub fn deinit(self: *Pool) void {
+        if (comptime leak_enabled) {
+            self.leak_detector.reportLeaks();
+            self.leak_detector.deinit();
+            self.allocator.destroy(self.leak_detector);
+        }
```

Comment:
If I'm reading this correctly, the leak detector will panic when the pool is deinitialized with nodes still active.

Keep in mind the eventual usage pattern for this pool will be:
- on startup
  - init
  - if possible, read a serialized pool from last process run
- during process run
  - create / ref / unref a bunch of stuff
- on shutdown
  - write the contents of the pool, serialized, to disk
  - deinit

Reply:
If that is the case, it seems hard to tell true leak or it should alive? Any thought/suggestion for that?